### PR TITLE
expand on repo structure behaviors and tradeoffs

### PR DESCRIPTION
**Pull request by armchairlinguist**
This is a great page, but customers were having trouble understanding the exact behavior and tradeoffs of each option, so I made a few things more explicit, and also made sure we used the same set of environment names throughout.

The line diff is a little annoying because I also re-justified everything (the alternative was weird inconsistent line lengths). I recommend the rich diff.

### CODE REVIEW

File: content/source/docs/enterprise/workspaces/repo-structure.html.md [hunk 1]
```diff
@@ -9,50 +9,27 @@ sidebar_current: "docs-enterprise2-workspaces-repo-structure"
...
-The user can then merge the PR and apply it in one workspace at a time, first with
-`networking-uat`, then `networking-stage`, and finally `networking-prod`. Eventually
-Terraform Enterprise will have functionality to enforce the stages in this pipeline.
-
-This model does not work if there are major environmental differences. For example, if the
-`networking-prod` workspace has 10 more unique resources than the `networking-stage` workspace,
-they likely cannot share the same Terraform configuration and thus cannot share the same repo.
-If there are major structural differences between environments, one of the below approaches
-can work.
+Using a single repo attached to multiple workspaces is the overall recommended approach, as it enables the creation of a pipeline to promote changes through environments, without additional overhead in version control. When using this model, one repo, such as `terraform-networking`, is connected to multiple workspaces — `networking-prod`, `networking-stage`, `networking-dev`. While the repo connection is the same in each case, each workspace can have a unique set of variables to configure the differences per environment.
```
Comment (maryelizbeth) [inline content/source/docs/enterprise/workspaces/repo-structure.html.md]: maybe sub `is the over all recommended approach` with `is generally considered best practice" 

just ruminating here --  im not too attached to either wording

File: content/source/docs/enterprise/workspaces/repo-structure.html.md [hunk 2]
```diff
@@ -72,25 +49,15 @@ workspace — one set of variables for `prod` and one set for `stage`.
 │   │   ├── outputs.tf
 ```
 
-Each workspace listens to a specific
-branch for changes, as configured by the [VCS branch setting](./settings.html#vcs-branch).
-The `networking-prod` workspace should be configured to listen to the `prod`
-branch and `networking-stage` to `stage.` To promote a change to stage, just open
-a PR against that branch. To promote to prod, open a PR for stage against prod.
+Each workspace listens to a specific branch for changes, as configured by the [VCS branch setting](./settings.html#vcs-branch). This means that plans will not occur in a given workspace until a PR is opened against or merged into the designated branch. The `networking-prod` workspace would be configured to listen to the `prod` branch, `networking-stage` to `stage`, and `networking-dev` to `dev`. To promote a change to `stage`, just open a PR against the `stage` branch. To promote to prod, open a PR from `stage` against `prod`.
```
Comment (maryelizbeth) [inline content/source/docs/enterprise/workspaces/repo-structure.html.md]: plans will also occur for `commit push` or `tag push` to the source branch (we can use `push event` or `push` to grab these wholesale)

Comment (maryelizbeth) [inline content/source/docs/enterprise/workspaces/repo-structure.html.md]: `To promote a change to stage, just` ... consider dropping `just`

Comment (armchairlinguist) [reply content/source/docs/enterprise/workspaces/repo-structure.html.md]: How does "until a PR is opened or a push event occurs" sound?

File: content/source/docs/enterprise/workspaces/repo-structure.html.md (hunk 1 above)
Comment (armchairlinguist) [reply content/source/docs/enterprise/workspaces/repo-structure.html.md]: I'm not quite sure either. The way I read the current phrase, it's saying "this is recommended in most cases, unless there are contraindications". "best practice" sounds stronger to me than that. But I also agree the "overall recommended approach" phrase is a bit strange.

Maybe something like "is the simplest best-practice approach"?

Comment (maryelizbeth) [reply content/source/docs/enterprise/workspaces/repo-structure.html.md]: 👍 on your new phrase.

**Pull request merged**